pvnet/models/multimodal/validation/config_static_validation.py

```diff
@@ -0,0 +1,514 @@
+                      f"sub-key: '_target_'"
+                  )
+
+    return section_content
+
+
+def _check_time_parameter(
+    cfg: dict[str, Any],
+    param_name: str,
+    owner_key: str,
+    required_if_owner_present: bool = True,
+) -> None:
+    """Check for a time parameter if its associated feature key is present/truthy.
+
+    Args:
+        cfg: The main configuration dictionary.
+        param_name: Name of the time parameter key (e.g., 'sat_history_minutes').
+        owner_key: Key indicating the associated feature is enabled (e.g., 'sat_encoder').
+        required_if_owner_present: If True, raise error if param_name is missing.
+
+    Raises:
+        KeyError: If `required_if_owner_present` is True and the parameter is missing.
+    """
+
+    # Only if associated feature / encoder is configured and enabled
+    if owner_key in cfg and cfg.get(owner_key):
+        context = f"Config includes '{owner_key}'"
+        if param_name not in cfg:
+            message = f"{context} but is missing '{param_name}'."
+            if required_if_owner_present:
+                raise KeyError(message)
+            else:
+                logger.warning(
+                    f"{message} (Action: May use default value if applicable)."
+                )
+        else:
+            _check_key(
+                cfg,
+                param_name,
+                required=False,
+                expected_type=int,
+                warn_on_type_mismatch=True,
+                context=f"Parameter '{param_name}' associated with '{owner_key}'",
+            )
+
+
+def _validate_nwp_specifics(cfg: dict[str, Any], nwp_section: dict[str, Any]) -> None:
+    """Validate NWP-specific time/interval dictionaries against NWP sources.
+
+    Args:
+        cfg: The main configuration dictionary.
+        nwp_section: The validated 'nwp_encoders_dict' section content.
+
+    Raises:
+        KeyError: If required NWP time/interval parameter dictionaries are missing.
+        TypeError: If NWP time/interval parameters are not dictionaries.
+        ValueError: If keys in time/interval dicts do not match NWP sources.
+    """
+    nwp_sources: list[str] = list(nwp_section.keys())
+    if not nwp_sources:
+        logger.warning("'nwp_encoders_dict' is defined but contains no NWP sources.")
+        return
+
+    context = f"Config includes 'nwp_encoders_dict' with sources: {nwp_sources}"
+
+    # Check time param values are integers
+    _check_key(
+        cfg, "nwp_history_minutes", required=True, expected_type=dict, context=context
+    )
+    _check_key(
+        cfg, "nwp_forecast_minutes", required=True, expected_type=dict, context=context
+    )
+    _check_key(
+        cfg, "nwp_interval_minutes", required=True, expected_type=dict, context=context
+    )
+
+    nwp_hist_times: dict[str, Any] = cfg["nwp_history_minutes"]
+    nwp_forecast_times: dict[str, Any] = cfg["nwp_forecast_minutes"]
```

Comment:
```suggestion
    nwp_hist_times: dict[str, int] = cfg["nwp_history_minutes"]
    nwp_forecast_times: dict[str, int] = cfg["nwp_forecast_minutes"]
```
I thought they were only allowed to be ints? Correct me if I'm wrong!